src/hooks/mediaHook.jsx

```diff
@@ -4,14 +4,14 @@ export default function useMobile(minWidth = "768px", boolean = true) {
     const [mobile, setMobile] = useState(boolean)
 
     useEffect(() => {
-        const media = window.matchMedia(`(min-width: ${minWidth})`)
+        const media = window.matchMedia?.(`(min-width: ${minWidth})`) || {}
```

Comment:
I don't like the idea of a useless object - how about leaving `media` as `undefined` if the method doesn't exist, then `if(!media) return` to exit early?

Reply:
I removed the empty fallback object and wrapped the effect body in a condition that tests for the `typeof` to ascertain if `window.matchMedia` is a `"function"`. This approach covers both a non-existent `window` and an undefined property of an existing `window`.